File: 41firstMissingPositive/firstMissingPositive.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var firstMissingPositive = function(nums) {
    for (let index = 0; !Object.is(nums.length, index); ++index)
        while (0 <= nums[index] - 1 && nums[index] - 1 < nums.length && !Object.is(nums[nums[index] - 1], nums[index])) [nums[nums[index] - 1], nums[index]] = [nums[index], nums[nums[index] - 1]]
    for (let index = 0; !Object.is(nums.length, index); ++index)
        if (!Object.is(nums[index], index + 1)) return index + 1
    return nums.length + 1
};

File: 41firstMissingPositive/firstMissingPositive.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        for (std::size_t index{0}; index != std::size(nums); ++index)
            while (0 <= nums.at(index) - 1 && nums.at(index) - 1 < std::size(nums) && nums.at(nums.at(index) - 1) != nums.at(index)) std::swap(nums.at(nums.at(index) - 1), nums.at(index));
        for (std::size_t index{0}; index != std::size(nums); ++index)
            if (nums.at(index) != index + 1) return index + 1;
        return std::size(nums) + 1;
    }
};
